test-data/unit/check-type-aliases.test

```diff
@@ -426,3 +426,95 @@ reveal_type(D.meth(1))  # E: Revealed type is 'Union[__main__.D*, builtins.int]'
 reveal_type(D().meth(1))  # E: Revealed type is 'Union[__main__.D*, builtins.int]'
 [builtins fixtures/classmethod.pyi]
 [out]
+
+[case testFlexibleAlias1]
+from typing import TypeVar, List, Tuple
+from mypy_extensions import FlexibleAlias
+
+T = TypeVar('T')
+U = TypeVar('U')
+
+AnInt = FlexibleAlias[T, int]
+
+x: AnInt[str]
+reveal_type(x)  # E: Revealed type is 'builtins.int'
+
+TwoArgs = FlexibleAlias[Tuple[T, U], bool]
+TwoArgs2 = FlexibleAlias[Tuple[T, U], List[U]]
+
+def welp(x: TwoArgs[str, int]) -> None:
+    reveal_type(x)  # E: Revealed type is 'builtins.bool'
+
+def welp2(x: TwoArgs2[str, int]) -> None:
+    reveal_type(x)  # E: Revealed type is 'builtins.list[builtins.int]'
+
+
+Id = FlexibleAlias[T, T]
+
+def take_id(x: Id[int]) -> None:
+    reveal_type(x)  # E: Revealed type is 'builtins.int'
+
+def id(x: Id[T]) -> T:
+    return x
+
+# TODO: This doesn't work and maybe it should?
+# Indirection = AnInt[T]
+# y: Indirection[str]
+# reveal_type(y)  # E : Revealed type is 'builtins.int'
```

Comment:
This doesn't work because all alias expansions happen immediately. When mypy sees `AnInt[T]` wherever, including in the r.h.s. above, it is immediately expanded to just `int`, and then a new alias is created, i.e. aliases are never "chained".

One can argue that this could be similar to how we allow omitting `Generic[T]` when subclassing `Iterable[T]`, but I would rather keep this simple, if someone really wants, one can use the form below.

Also it would be a relatively larger change because we would need to move querying for type variables in the middle of analyzing the alias.